.rubocop.yml

```diff
@@ -266,6 +266,15 @@ Style/SymbolArray:
   # MO uses plain brackets. Ex: [:foo, :bar, :baz]
   EnforcedStyle: brackets
 
+Rails/ActionOrder:
+  ExpectedOrder:
+    - index
+    - show
+    - new
+    - create
+    - edit
+    - update
+    - destroy
 Rails/SkipsModelValidations:
   # We skip validations all the time.  I'm tired of disabling this cop.
   Enabled: false
```

Comment:
Suggestions:
1. Move these settings up to where the file has the other settings Rails department cops. (The out-of place setting might be my fault.)
2. Note that we have two configurations for `Rails/ActionOrder`. They should be combined, at least in this file. Otherwise we get this warning when running RuboCop:
>.rubocop.yml:159: `Rails/ActionOrder` is concealed by line 269

3. I prefer the settings above, which comment out `edit` because that's more flexible, resulting in no offenses.
If we demand that `create` appear before `edit`, we end up with 6 offenses, which should be fixed.  (With the default order there are 8 offenses.)
>920 files inspected, 6 offenses detected:
[app/controllers/articles_controller.rb - 1 offense](https://github.com/MushroomObserver/mushroom-observer/pull/1254/files#offense_app/controllers/articles_controller.rb)
[app/controllers/glossary_terms_controller.rb - 1 offense](https://github.com/MushroomObserver/mushroom-observer/pull/1254/files#offense_app/controllers/glossary_terms_controller.rb)
[app/controllers/herbaria_controller.rb - 1 offense](https://github.com/MushroomObserver/mushroom-observer/pull/1254/files#offense_app/controllers/herbaria_controller.rb)
[app/controllers/publications_controller.rb - 1 offense](https://github.com/MushroomObserver/mushroom-observer/pull/1254/files#offense_app/controllers/publications_controller.rb)
[app/controllers/visual_groups_controller.rb - 1 offense](https://github.com/MushroomObserver/mushroom-observer/pull/1254/files#offense_app/controllers/visual_groups_controller.rb)
[app/controllers/visual_models_controller.rb - 1 offense](https://github.com/MushroomObserver/mushroom-observer/pull/1254/files#offense_app/controllers/visual_models_controller.rb)

Options for fixing include:
- change the action order in the offending files. Issues:
  -  I deliberately use the default order
```ruby
 # ---------- Actions to Display forms -- (new, edit, etc.) -------------------

  def new
...
  def edit
...
  # ---------- Actions to Modify data: (create, update, destroy, etc.) ---------
  def create
...
  def update
...
```
  - Based on the `visual` stuff, Nathan (and other devs) will probably naturally choose the default order.
- disable the cop inline in the offending files.
- add additional .rubocop.yml files in the appropriate directories.

Reply:
I think i did that provisionally, when you first added the cop. Will remove

I have to say that with these fat controllers I'm working on, I can't understand why anyone would prefer the default order. It's great that the new Visual controllers are skinny, but for all the other CRUD controllers i've had to hack through, if you read through them, I think you will find it is way too much scrolling to try to follow logic from `new` to `create`, via `edit`.  Moving the private methods to the end doesn't seem helpful either — I tried that. (I don't know, even though I have a scroll mouse i have lost a lot of time and frustration just through scrolling, and I didn't pick this organization without experimentation.)

When we move to service objects it won't matter though, because they will be so slim that skimming the controller methods won't even require scrolling.